### PR TITLE
HIVE-27417: Remove Filter.g and unnecessary antlr3 dependency

### PR DESCRIPTION
**Pull request by wecharyu**
### What changes were proposed in this pull request?
1. Remove ANTLR3 grammar file `Filter.g`
2. Remove some unused antlr3 dependency in pom modules

### Why are the changes needed?
Since we introduced ANTLR4 grammar file `PartitionFilter.g4`, it's safe to remove `Filter.g` now.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass all existing tests.

### CODE REVIEW

**Review by aturoczy — APPROVED** [review]
LGTM

**Comment by aturoczy** [comment]
@veghlaci05 @kasakrisz Could you pls check?

File: standalone-metastore/pom.xml [hunk 1]
```diff
@@ -57,7 +57,6 @@
     <maven.versions.plugin.version>2.13.0</maven.versions.plugin.version>
     <maven.surefire.plugin.version>3.0.0-M4</maven.surefire.plugin.version>
     <!-- Dependency versions -->
-    <antlr.version>3.5.2</antlr.version>
     <antlr4.version>4.9.3</antlr4.version>
```
[the remark this answers is not in the document]
Comment (wecharyu) [reply standalone-metastore/pom.xml]: Done.

**Review by kasakrisz — APPROVED** [review]
+1 pending tests

**Pull request merged**